### PR TITLE
SNTWREC-27: Set up CICD for backend microservices

### PR DESCRIPTION
**Pull request by nwittenberg**
- add CICD for backend services

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -61,58 +61,98 @@ jobs:
...
+  build_publish_deploy_ingest_dev:
     needs: unit_tests
-    name: docker_build_publish_dev
-    runs-on: ubuntu-latest
+    uses: ./.github/workflows/build-publish-deploy.yml
     if: github.event_name == 'push' && github.ref == 'refs/heads/dev'
+    with:
+      service-name: ingest
+      stage: dev
+    secrets:
```
Comment (grusc) [inline .github/workflows/main.yml:72]: using `secrets: inherit` would be less verbose. See: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-of-jobsjob_idsecretsinherit
Any reason we want to pass secrets explicitly?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -61,58 +61,98 @@ jobs:
           docker build --file ./microservices/embeddingservice/docker/Dockerfile ./microservices/embeddingservice
           echo "embedding service image build OK"
 
-  
-  docker_build_publish_dev:
+
+  build_publish_deploy_ingest_dev:
     needs: unit_tests
```
Comment (grusc) [inline .github/workflows/main.yml:66]: Depending on reco explorer unit test does not make sense to me.
Wouldn't we want to run microservice unit tests here?

**Review by grusc — COMMENTED** [review]
We potentially doing more work than we need to, as we always deploy all services even if there are no changes in some of them.

However if we still want to do that, using a matrix strategy makes the pipeline much less verbose. See: https://docs.github.com/de/actions/using-jobs/using-a-matrix-for-your-jobs#using-a-matrix-strategy

File: .github/workflows/main.yml (hunk 2 above)
Comment (nwittenberg) [reply .github/workflows/main.yml]: good catch

**Comment by nwittenberg** [comment]
@grusc thx for your comments - I redesigned the approach to accommodate builds on changes only - makes sense to me.

**Comment by grusc** [comment]
LGTM
very nice!

**Pull request merged**